overlord/configstate/configcore/tmp.go

```diff
@@ -0,0 +1,182 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"syscall"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/sysconfig"
+)
+
+const (
+	mntStaticOptions         = "mode=1777,strictatime,nosuid,nodev"
+	tmpfsMountPoint          = "/tmp"
+	tmpMntServOverrideSubDir = "tmp.mount.d"
+	tmpMntServOverrideFile   = "override.conf"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.tmp.size"] = true
+}
+
+// Regex matches what is specified by tmpfs(5) for the size option
+var validTmpfsSizeRe = regexp.MustCompile(`^[0-9]+[kmgKMG%]?$`).MatchString
+
+func validTmpfsSize(sizeStr string) error {
+	if !validTmpfsSizeRe(sizeStr) {
```

Comment:
Please check `gadget/quantity/size.go` I think there is some code that can be reused from there.

Reply:
Sounds good. But, mount options also allow to specify a percentage of the physical memory, is it ok to parse that as a special case for tmp.size?

Reply:
let's not support percentages for now, we have other things which need to work with percentages that we are going to have a design meeting with Gustavo on Monday about (cpu quotas) and so after that meeting we may have better guidance on the direction to take with respect to implementing percentages